src/snarky/js-layout.ts

```diff
@@ -0,0 +1,2344 @@
+export { jsLayout };
```

Comment:
can you put this in a `gen/` folder (I'm assuming this is autogenerated). I think we should still check it in so we can codereview when it changes etc

Reply:
what do we need to do in CI to make sure a test fails if the layout changes and it breaks something in snarkyjs?

Reply:
I tried to make sure that, if the layout changes, either:
* 1. snarkyjs continues working, because everything that needs to change changes automatically, or
* 2. there's a type error in OCaml, or
* 3. there's a type error in TS

I think 2.) is the weakest point at the moment, because in some places we rely on strings in OCaml as hints to what JS types should be, and I want to reduce that further in the future.

I tried to ensure 3.) as much as possible; in particular, the manually written conversion functions for toJson / toFields are type-checked.

If there's a type error in TS, CI would fail pretty early, both in the snarkyjs unit test and in the "build JS integration test" job